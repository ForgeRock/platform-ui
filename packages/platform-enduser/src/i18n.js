/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
