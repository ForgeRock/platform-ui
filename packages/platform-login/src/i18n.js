/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
