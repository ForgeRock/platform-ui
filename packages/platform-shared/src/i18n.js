/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
