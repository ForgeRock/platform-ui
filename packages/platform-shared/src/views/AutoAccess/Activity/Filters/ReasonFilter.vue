<!-- Copyright (c) 2022-2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div>
    <FrField
      :options="reasonOptions"
      label="Reason"
      type="multiselect"
      v-model="selected"
    />
  </div>
</template>
<script>
import FrField from '@forgerock/platform-shared/src/components/Field/';
import store from '@/store';
import { causeMap } from '../api/ActivityAPI';

export default {
  name: 'ReasonFilter',
  components: {
    FrField,
  },
  props: {
    value: {
      default: () => [],
      type: Array,
    },
  },
  computed: {
    selected: {
      get() {
        return this.value;
      },
      set(value) {
        this.$emit('update', value);
      },
    },
    /**
     * Formats a list of options used in a multiselect filter
     *
     * @returns {Array} filter options
     */
    reasonOptions() {
      const defaultOptions = [
        'is_advanced_bot_detection',
        'is_automated_user_agent',
        'is_brute_force',
        'is_credential_stuffing',
        'is_distributed_attack',
        'is_impossible_travel',
        'is_ip_blocked',
        'is_suspicious_ip',
      ];
      const uebaList = store.state.Dashboard.uebaClusteringReasons.map((reason) => (
        {
          text: `Unusual ${causeMap[reason]}`,
          value: reason,
        }
      ));
      const uebaOptions = [];
      uebaList.forEach((item) => {
        if (!uebaOptions.some((option) => option.text === item.text)) {
          const list = [];
          uebaList.forEach((reason) => {
            if (reason.text === item.text) {
              list.push(reason.value);
            }
          });
          uebaOptions.push({
            text: item.text,
            value: list,
          });
        }
      });
      return [
        ...defaultOptions.map((heuristic) => ({ text: this.$t(`autoAccess.access.reasons["${heuristic}"]`), value: [heuristic] })),
        ...uebaOptions,
      ];
    },
  },
};
</script>
