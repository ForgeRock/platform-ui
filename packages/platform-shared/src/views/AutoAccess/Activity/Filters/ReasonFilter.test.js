/**
 * Copyright (c) 2022-2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import ReasonFilter from './ReasonFilter';
import i18n from '@/i18n';

jest.mock('@/store', () => ({
  __esModule: true,
  default: {
    state: {
      Dashboard: {
        uebaClusteringReasons: [
          'browser',
          'user_agent',
          'city',
          'country',
          'day_of_week',
          'weekday',
          'model',
          'device',
          'os',
          'os_version',
          'osWithVersion',
          'dayparting',
        ],
      },
    },
  },
}));

describe('Reason Filter', () => {
  let wrapper;
  const expectedOptions = [
    {
      text: 'Advanced Bot Detection',
      value: ['is_advanced_bot_detection'],
    },
    {
      text: 'Automated User Agent',
      value: ['is_automated_user_agent'],
    },
    {
      text: 'Brute Force',
      value: ['is_brute_force'],
    },
    {
      text: 'Credential Stuffing',
      value: ['is_credential_stuffing'],
    },
    {
      text: 'Distributed Attack',
      value: ['is_distributed_attack'],
    },
    {
      text: 'Impossible Travel',
      value: ['is_impossible_travel'],
    },
    {
      text: 'IP Blocklist',
      value: ['is_ip_blocked'],
    },
    {
      text: 'Suspicious IP',
      value: ['is_suspicious_ip'],
    },
    {
      text: 'Unusual Browser',
      value: ['browser'],
    },
    {
      text: 'Unusual Browser Family',
      value: ['user_agent'],
    },
    {
      text: 'Unusual City',
      value: ['city'],
    },
    {
      text: 'Unusual Country',
      value: ['country'],
    },
    {
      text: 'Unusual Day of Week',
      value: ['day_of_week', 'weekday'],
    },
    {
      text: 'Unusual Device',
      value: ['model', 'device'],
    },
    {
      text: 'Unusual OS',
      value: ['os'],
    },
    {
      text: 'Unusual OS Version',
      value: ['os_version', 'osWithVersion'],
    },
    {
      text: 'Unusual Time of Day',
      value: ['dayparting'],
    },
  ];

  beforeEach(() => {
    wrapper = shallowMount(ReasonFilter, {
      global: {
        plugins: [i18n],
      },
    });
  });
  it('Component should render', () => {
    expect(wrapper.vm.reasonOptions).toEqual(expectedOptions);
  });
});
