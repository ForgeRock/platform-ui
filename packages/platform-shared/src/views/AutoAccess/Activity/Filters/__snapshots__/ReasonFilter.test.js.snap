// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reason Filter Component should render: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <FrField",
    "options=",
    [
      {
        "text": "Advanced Bot Detection",
        "value": [
          "is_advanced_bot_detection",
        ],
      },
      {
        "text": "Automated User Agent",
        "value": [
          "is_automated_user_agent",
        ],
      },
      {
        "text": "Brute Force",
        "value": [
          "is_brute_force",
        ],
      },
      {
        "text": "Credential Stuffing",
        "value": [
          "is_credential_stuffing",
        ],
      },
      {
        "text": "Distributed Attack",
        "value": [
          "is_distributed_attack",
        ],
      },
      {
        "text": "Impossible Travel",
        "value": [
          "is_impossible_travel",
        ],
      },
      {
        "text": "IP Blocklist",
        "value": [
          "is_ip_blocked",
        ],
      },
      {
        "text": "Suspicious IP",
        "value": [
          "is_suspicious_ip",
        ],
      },
      {
        "text": "Unusual Browser",
        "value": [
          "browser",
        ],
      },
      {
        "text": "Unusual Browser Family",
        "value": [
          "user_agent",
        ],
      },
      {
        "text": "Unusual City",
        "value": [
          "city",
        ],
      },
      {
        "text": "Unusual Country",
        "value": [
          "country",
        ],
      },
      {
        "text": "Unusual Day of Week",
        "value": [
          "day_of_week",
          "weekday",
        ],
      },
      {
        "text": "Unusual Device",
        "value": [
          "model",
          "device",
        ],
      },
      {
        "text": "Unusual OS",
        "value": [
          "os",
        ],
      },
      {
        "text": "Unusual OS Version",
        "value": [
          "os_version",
          "osWithVersion",
        ],
      },
      {
        "text": "Unusual Time of Day",
        "value": [
          "dayparting",
        ],
      },
    ],
    "label="Reason"",
    "type="multiselect"",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "canEnterPlaceholders" with v-bind value \`false\` will render canEnterPlaceholders="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <FrField",
    "options=",
    [
      {
        "text": "Advanced Bot Detection",
        "value": [
          "is_advanced_bot_detection",
        ],
      },
      {
        "text": "Automated User Agent",
        "value": [
          "is_automated_user_agent",
        ],
      },
      {
        "text": "Brute Force",
        "value": [
          "is_brute_force",
        ],
      },
      {
        "text": "Credential Stuffing",
        "value": [
          "is_credential_stuffing",
        ],
      },
      {
        "text": "Distributed Attack",
        "value": [
          "is_distributed_attack",
        ],
      },
      {
        "text": "Impossible Travel",
        "value": [
          "is_impossible_travel",
        ],
      },
      {
        "text": "IP Blocklist",
        "value": [
          "is_ip_blocked",
        ],
      },
      {
        "text": "Suspicious IP",
        "value": [
          "is_suspicious_ip",
        ],
      },
      {
        "text": "Unusual Browser",
        "value": [
          "browser",
        ],
      },
      {
        "text": "Unusual Browser Family",
        "value": [
          "user_agent",
        ],
      },
      {
        "text": "Unusual City",
        "value": [
          "city",
        ],
      },
      {
        "text": "Unusual Country",
        "value": [
          "country",
        ],
      },
      {
        "text": "Unusual Day of Week",
        "value": [
          "day_of_week",
          "weekday",
        ],
      },
      {
        "text": "Unusual Device",
        "value": [
          "model",
          "device",
        ],
      },
      {
        "text": "Unusual OS",
        "value": [
          "os",
        ],
      },
      {
        "text": "Unusual OS Version",
        "value": [
          "os_version",
          "osWithVersion",
        ],
      },
      {
        "text": "Unusual Time of Day",
        "value": [
          "dayparting",
        ],
      },
    ],
    "label="Reason"",
    "type="multiselect"",
    " ...",
    ">",
    "
",
    " at <ReasonFilter",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
